### PR TITLE
Feature/update testing env and replace enzyme by testing library react

### PR DESCRIPTION
**Pull request by Kelvur**
The test actually are broken, @testing-library/react don't cleanup after a test and the test are pollutin other test. You can observate this behaviour testing `I18nProvider`.

### CODE REVIEW

**Comment by rubeniskov** [comment]
## Problem detection 

Ava provides a debug inspector (using the native NodeJS --inpsect-brk) this would help you to find the source of the problem.

Here you have an example

```shell
NODE_ENV=test ./node_modules/.bin/ava debug src/I18nProvider.spec.js 
````

Now you can open your developer tools in chrome and click in NodeJS Inspector utility

image

output

> gif created using the following command of ffmpeg
```shell
ffmpeg -i debug_with_ava.mov -vf "fps=10,scale=720:-1:flags=lanczos,split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse" -loop 0 output.gif
```


After inspecting the logic of cleanup, the behaviour is async so cleanup returns a promise to handle when the unmouting/mounting cycle is complete.

> Before
image

> After 
image
 


## Current solution

In this case due to the nature of the runner which execute each test concurrently, you have to indicate (in this case of having this global state) which tests you want to execute in series (secuentialy) in order to preserve the cleanup order corretly.

image


Now all the test that can be executed concurrently will be behave has before but the specs such need manage this global state will execute in serial resulting the expected behaviour.

image
 

 ```patch
diff --git a/src/I18nProvider.spec.js b/src/I18nProvider.spec.js
index ab022c7..b540441 100644
--- a/src/I18nProvider.spec.js
+++ b/src/I18nProvider.spec.js
@@ -10,19 +10,19 @@ import I18nProvider from './I18nProvider';
 
 test.afterEach(cleanup);
 
-test('render without errors', async (t) => {
+test.serial('render without errors', async (t) => {
   const i18n = await fetchI18n();
   render(<I18nProvider i18n={i18n}><Consumer /></I18nProvider>);
   t.pass();
 });
 
-test('translates correctly', async (t) => {
+test.serial('translates correctly', async (t) => {
   const i18n = await fetchI18n();
   const { getByText } = render(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
   t.truthy(getByText('Phone'));
 });
 
-test('updates correctly after setLocale', async (t) => {
+test.serial('updates correctly after setLocale', async (t) => {
   const i18n = await fetchI18n();
   const { getByText, rerender } = render(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
   t.truthy(getByText('Phone'));
```


## Better approach

Other solution to keep the concurretly behaviour and avoid the cleanup sequence is use a [customRender](https://testing-library.com/docs/react-testing-library/setup#custom-render) without a global state

```diff
diff --git a/src/I18nProvider.spec.js b/src/I18nProvider.spec.js
index ab022c7..60dc4c0 100644
--- a/src/I18nProvider.spec.js
+++ b/src/I18nProvider.spec.js
@@ -2,29 +2,35 @@
 import React from 'react';
 // Testing
 import test from 'ava';
-import { render, cleanup } from '@testing-library/react';
+import { render } from '@testing-library/react';
 import { fetchI18n, Consumer } from './testUtils';
 // Components
 import I18nProvider from './I18nProvider';
 
 
-test.afterEach(cleanup);
+const customRender = (node, {
+  container = document.createElement('div'),
+  ...rest
+} = {}) => render(node, {
+  container,
+  ...rest,
+});
 
 test('render without errors', async (t) => {
   const i18n = await fetchI18n();
-  render(<I18nProvider i18n={i18n}><Consumer /></I18nProvider>);
+  customRender(<I18nProvider i18n={i18n}><Consumer /></I18nProvider>);
   t.pass();
 });
 
 test('translates correctly', async (t) => {
   const i18n = await fetchI18n();
-  const { getByText } = render(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
+  const { getByText } = customRender(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
   t.truthy(getByText('Phone'));
 });
 
 test('updates correctly after setLocale', async (t) => {
   const i18n = await fetchI18n();
-  const { getByText, rerender } = render(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
+  const { getByText, rerender } = customRender(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
   t.truthy(getByText('Phone'));
   await i18n.setLocale('de');
   rerender(<I18nProvider i18n={i18n}><Consumer word="Teléfono" /></I18nProvider>);
```

**Comment by Kelvur** [comment]
I just sent the last modifications, I think this branch it's ready for merge @rubeniskov

File: src/I18nContext.spec.js [hunk 1]
```diff
@@ -1,53 +1,15 @@
 // Core
 import React from 'react';
-import ReactDOM from 'react-dom';
-// Test
-import {act} from 'react-dom/test-utils';
+// Testing
 import test from 'ava';
+import render from '@test/customRender';
```
Comment (rubeniskov) [inline src/I18nContext.spec.js:5]: I would like see this more likely the @testing-library/react, using the same api but including a new method

```javascript
import { customRender, cleanup, ...(rest of the original utilities) } from '@test/react';
````

**Review by rubeniskov — APPROVED** [review]
I leave this for a future approach

File: test/testUtils.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      'esto es una prueba': 'Das ist ein Test',
+      Teléfono: 'Telefon',
+      Perro: 'Hund',
+    },
+  },
+};
+
+export function fetchI18n() {
+  return new Promise((resolve) => {
+    const i18n = i18njs(I18N_DATA);
```
Comment (rubeniskov) [inline test/testUtils.js:32]: I will leave a issue to change this in a future by the newer release of `i18njs` [0.2.0](https://github.com/k14v/i18njs/pull/21)

**Comment by rubeniskov** [comment]
Nice coverage!! congrats LGTM I will let you release the new version ;)

**Pull request merged**